federation/pkg/federation-controller/service/dns.go

```diff
@@ -89,23 +90,31 @@ func (s *ServiceController) getClusterZoneNames(clusterName string) (zones []str
 	return client.cluster.Status.Zones, client.cluster.Status.Region, nil
 }
 
-// getFederationDNSZoneName returns the name of the managed DNS Zone configured for this federation
-func (s *ServiceController) getFederationDNSZoneName() (string, error) {
-	return s.zoneName, nil
+// getFederationDnsSuffix returns the DNS suffix to use for records for this federation
+func (s *ServiceController) getFederationDnsSuffix() (string, error) {
+	return s.federationDnsSuffix, nil
 }
 
-// getDnsZone is a hack around the fact that dnsprovider does not yet support a Get() method, only a List() method.  TODO:  Fix that.
-func getDnsZone(dnsZoneName string, dnsZonesInterface dnsprovider.Zones) (dnsprovider.Zone, error) {
-	dnsZones, err := dnsZonesInterface.List()
+// getDnsZone returns the zone, as identified by zoneName
+func (s *ServiceController) getDnsZone() (dnsprovider.Zone, error) {
```

Comment:
I don't really like making this is member function of ServiceController.  It has nothign to do with service controllers. If anything it should be a member of DNSZone, or a util function.  Can we revert the function signature back to what it was?